pkg/certificate/providers/config_test.go

```diff
@@ -155,52 +287,67 @@ func TestGetCertificateManager(t *testing.T) {
 				VaultProtocol: "hi",
 			},
 			expectError: true,
+			cfg:         mockConfigurator,
```

Comment:
additionally mockConfigurator, kubeclient, and configclient can all be factored out.

To remove the latter, we can instead have on the testcase struct, a slice of `MeshRootCertificate`s